include/lib.h

```diff
@@ -144,11 +145,42 @@ static inline uint64_t rdmsr(uint32_t msr_idx) {
     return (((uint64_t) high) << 32) | low;
 }
 
+/*
+ * read the msr_idx msr; the result is valid iff *read is true
+ */
+static inline uint64_t rdmsr_safe(uint32_t msr_idx, bool *read) {
```

Comment:
nit: I would invert the logic here: the function can return `int` and set the value to input parameter. Same for `wrmsr_safe`.